Sources/VSM/StateContainer.swift

```diff
@@ -37,21 +69,20 @@ final public class StateContainer<State>: ObservableObject {
 
 public extension StateContainer {
     /// Observes the state publisher emitted as a result of invoking some action
-    func observe(_ stateChangePublisher: AnyPublisher<State, Never>) {
+    func observe(_ statePublisher: AnyPublisher<State, Never>) {
         cancelRunningObservations()
-        cancellable = stateChangePublisher
-            .receive(on: DispatchQueue.main)
+        stateSubscription = statePublisher
             .sink { [weak self] newState in
-                self?.state = newState
+                self?.setStateOnMainThread(to: newState)
             }
     }
     
     /// Observes the state emitted as a result of invoking some asynchronous action
-    func observe(_ awaitState: @escaping () async -> State) {
+    func observeAsync(_ nextState: @escaping () async -> State) {
         cancelRunningObservations()
         // A weak-self declaration is required on the `Task` closure to break an unexpected strong self retention, despite not directly invoking self ¯\_(ツ)_/¯
-        asyncTask = Task(priority: .userInitiated) { [weak self] in
-            let newState = await awaitState()
+        stateTask = Task(priority: .userInitiated) { [weak self] in
```

Comment:
I've come to learn that specifying a priority when creating a task means that your entire app must also be "priority aware" or you may start seeing QoS violations at runtime. It feels oh so righty right to indicate that this task here should execute with a `.userInitiated` priority, so I get it. I wonder if some kind of documentation about how the framework uses tasks with this priority and so developers should be aware of other QoS opportunities in their code (this includes Tasks, Operation Queues, and any of the threading patterns available.)

Reply:
That is a good point. The priority specification here is unnecessary. Do you think it's important enough to change as part of this PR or benign enough that we can defer this change to when we implement the actor system in our future iOS 15 support?

Reply:
I went ahead and removed it for now.

`@MainActor` is considered reliable for general use in iOS 15+. A future iOS 15+ VSM update will make that transition. One of the main benefits being the inherent main-thread optimization for async functions that turn out to be synchronous (ie, checking local cache and returning a cached value instead of loading from the server)